File: iosApp/iosApp/views/navigation/bars/Level1NavigationRail.swift
//
//  Level1NavigationRail.swift
//  iosApp
//
//  Created by Daniele Baroncelli on 24/05/21.
//  Copyright © 2021 orgName. All rights reserved.
//

import SwiftUI
import shared


// this is the left vertical navigation bar for 2-Pane visualization
// (used by bigger devices in landscape mode)
    
struct Level1NavigationRail: View {
    @EnvironmentObject var appObj: AppObservableObject
        
    var body: some View {
        let level1ScreenIdentifier = appObj.localNavigationState.currentLevel1ScreenIdentifier
        VStack {
            Spacer()
            NavigationRailButton(
                itemLabel: "All Countries",
                iconName: "list.bullet",
                selected: level1ScreenIdentifier.URI==Level1Navigation.allCountries.screenIdentifier.URI,
                onClick: { appObj.dkmpNav.navigateByLevel1Menu(appObj, level1Navigation: Level1Navigation.allCountries) }
            )
            NavigationRailButton(
                itemLabel: "Favorites",
                iconName: "star.fill",
                selected: level1ScreenIdentifier.URI==Level1Navigation.favoriteCountries.screenIdentifier.URI,
                onClick: { appObj.dkmpNav.navigateByLevel1Menu(appObj, level1Navigation: Level1Navigation.favoriteCountries) }
            )
            Spacer()
        }
    }
    
}




struct NavigationRailButton: View {
    var itemLabel : String
    var iconName : String
    var selected : Bool
    var onClick : () -> Void
    
    var body: some View {
        Button(action: { onClick() }) {
            VStack(spacing: 5) {
                Image(systemName: iconName).resizable().scaledToFit().frame(height:15)
                Text(itemLabel).font(Font.footnote)
            }
            .padding(.top, 25)
            .padding(.bottom, 25)
            .foregroundColor(selected ? .white : linkColor)
        }
    }
}

File: composables/src/commonMain/kotlin/eu/baroncelli/dkmpsample/composables/navigation/bars/Level1NavigationRail.kt
package eu.baroncelli.dkmpsample.composables.navigation.bars

import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Menu
import androidx.compose.material.icons.filled.Star
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.sp
import eu.baroncelli.dkmpsample.shared.viewmodel.Navigation
import eu.baroncelli.dkmpsample.shared.viewmodel.ScreenIdentifier
import eu.baroncelli.dkmpsample.shared.viewmodel.screens.Level1Navigation


// this is the left vertical navigation bar for 2-Pane visualization
// (used by bigger devices in landscape mode)

@Composable
fun Navigation.Level1NavigationRail(
    selectedTab: ScreenIdentifier,
    navigateByLevel1Menu: (Level1Navigation) -> Unit
) {
    NavigationRail(content = {
        Spacer(Modifier.weight(1f))
        NavigationRailItem(
            icon = { Icon(Icons.Default.Menu, "ALL") },
            label = { Text("All Countries", fontSize = 13.sp) },
            selected = selectedTab.URI == Level1Navigation.AllCountries.screenIdentifier.URI,
            onClick = { navigateByLevel1Menu(Level1Navigation.AllCountries) }
        )
        NavigationRailItem(
            icon = { Icon(Icons.Default.Star, "FAVORITES") },
            label = { Text("Favourites", fontSize = 13.sp) },
            selected = selectedTab.URI == Level1Navigation.FavoriteCountries.screenIdentifier.URI,
            onClick = { navigateByLevel1Menu(Level1Navigation.FavoriteCountries) }
        )
        Spacer(Modifier.weight(1f))
    }, modifier = Modifier.fillMaxSize(), containerColor = MaterialTheme.colorScheme.primaryContainer)
}
